src/vellum/workflows/nodes/bases/base.py

```diff
@@ -146,7 +146,19 @@ def __new__(mcs, name: str, bases: Tuple[Type, ...], dct: Dict[str, Any]) -> Any
         node_class.Execution.node_class = node_class
         node_class.Trigger.node_class = node_class
         node_class.ExternalInputs.__parent_class__ = node_class
-        node_class.__id__ = uuid4_from_hash(node_class.__qualname__)
+
+        # Check if this node uses legacy ID generation (qualname only) or new ID generation (module + qualname)
+        use_legacy_id = False
+        for base_class in node_class.__mro__:
```

Comment:
DevinAI Don't iterate over __mro__. It's a simple `if node_class.__legacy_id__` check